src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1155,6 +1157,28 @@ public IEnumerable<Tombstone> GetTombstonesFrom(DocumentsOperationContext contex
             }
         }
 
+        public GetTombstonesPreviewResult GetTombstonesPreviewResult(DocumentsOperationContext context, long etag, long start, long take)
+        {
+            var table = new Table(TombstonesSchema, context.Transaction.InnerTransaction);
+
+            var tombstones = new List<Tombstone>();
+            foreach (var result in table.SeekForwardFrom(TombstonesSchema.FixedSizeIndexes[AllTombstonesEtagsSlice], etag, start))
+            {
+                if (take-- <= 0)
+                    break;
+
+                var tombstone = TableValueToTombstone(context, ref result.Reader);
+                tombstones.Add(tombstone);
+            }
+
+            tombstones.Sort(ShardedReplicationHandlerProcessorForGetTombstones.TombstonesPreviewComparer.Instance);
```

Comment:
I do not like that we used here Sharded items. The preview comparer should be either extracted outside the processor or sorting should be done in the processor.

Reply:
Done on https://github.com/ravendb/ravendb/pull/14908